mlflow/lightgbm.py

```diff
@@ -837,22 +847,57 @@ def infer_model_signature(input_example):
                 registered_model_name=registered_model_name,
             )
 
+        # Whether to automatically log the training dataset as a dataset artifact.
+        if _log_datasets:
+            try:
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
+
+                _log_lightgbm_dataset(train_set, source, "train", autologging_client)
+
+                valid_sets = kwargs.get("valid_sets")
+                if valid_sets is not None:
+                    valid_names = kwargs.get("valid_names")
+                    if valid_names is None:
+                        for valid_set in valid_sets:
+                            _log_lightgbm_dataset(valid_set, source, "eval", autologging_client)
+                    else:
+                        for valid_set, valid_name in zip(valid_sets, valid_names):
+                            _log_lightgbm_dataset(
+                                valid_set, source, "eval", autologging_client, name=valid_name
+                            )
+
+                dataset_logging_operations = autologging_client.flush(synchronous=False)
+                dataset_logging_operations.await_completion()
+            except Exception as e:
+                _logger.warning(
+                    "Failed to log training dataset information to MLflow Tracking. Reason: %s", e
```

Comment:
```suggestion
                    "Failed to log dataset information to MLflow Tracking. Reason: %s", e
```

(Dataset logging failure could be training, evaluation, or both)